src/codeu/chat/client/commandline/Panel.java

```diff
@@ -17,42 +17,394 @@
 import java.util.HashMap;
 import java.util.Map;
 import java.util.Scanner;
+import java.util.Stack;
+import java.util.ArrayList;
+import java.util.List;
+import java.io.IOException;
 
-// PANEL
-//
-// A panel is a collection of commands that are to be executed within a specifc
-// context. Similar to how panels are used in a GUI, is is a command line
-// equivalent.
-//
-final class Panel {
-
-  public interface Command {
-    void invoke(Scanner line);
-  }
+import codeu.chat.client.core.Context;
+import codeu.chat.client.core.ConversationContext;
+import codeu.chat.client.core.MessageContext;
+import codeu.chat.client.core.UserContext;
+import codeu.chat.util.Tokenizer;
 
-  private final Map<String, Command> commands = new HashMap<>();
+public final class Chat {
```

Comment:
It looks like you copied all of the code `Chat` class into the `Panel.java` file. Was that intentional? I'd recommend changing this file back so that it only contains the updates necessary for the tokenization feature, and only contains the `Panel` public class.  All of the code in the `Chat` class should really stay in `Chat.java`.